core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6562,6 +6562,67 @@ private HepProgram getTransitiveProgram() {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6214">[CALCITE-6214]
+   * Remove `DISTINCT` in `COUNT` if field is unique</a>. */
+  @Test void testAggregateDistinctRemove1() {
+    final String sql = ""
+        + "select count(distinct x) cnt\n"
+        + "from(\n"
+        + "   select distinct sal x from emp\n"
+        + ") t ";
+    sql(sql)
+        .withRule(CoreRules.AGGREGATE_REMOVE_DISTINCT)
+        .check();
+  }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6214">[CALCITE-6214]
+   * Remove `DISTINCT` in `COUNT` if field is unique</a>. */
+  @Test void testAggregateDistinctRemove2() {
+    final String sql = ""
+        + "select count(distinct x) cnt\n"
+        + "from(\n"
+        + "   select sal * 12 as x from emp"
+        + "   group by sal * 12\n"
+        + ") t ";
+    sql(sql)
+        .withRule(CoreRules.AGGREGATE_REMOVE_DISTINCT)
+        .check();
+  }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6214">[CALCITE-6214]
+   * Remove `DISTINCT` in `COUNT` if field is unique</a>. */
+  @Test void testAggregateDistinctRemove3() {
```

Comment:
If you handle other aggregates you should probably add tests like `SELECT COUNT(x), SUM(x) FROM SELECT DISTINCT ...`

Reply:
Yes, now we can handle other aggregate functions after new commits.